docs/roles/integrator/integrating.md

```diff
@@ -0,0 +1,372 @@
+
+### Create key pair (offline)
+
+NEAR supports generating public/private key pairs from a random seed or using a string to generate a public key from a given private key.
+
+Generating key pairs two ways and comparing them
+
+```js
+// generating a key pair
+const keyPair = nearlib.utils.key_pair.KeyPairEd25519.fromRandom();
+
+// generating another key pair based on the private key of the key pair above
+const newKeyPair = nearlib.utils.key_pair.KeyPair.fromString(keyPair.toString());
+
+// the two will match
+console.assert(newKeyPair.secretKey === keyPair.secretKey);
+```
+
+Generating a key pair to sign and verify a message
+
+```js
+// generate a key pair
+const keyPair = nearlib.utils.key_pair.KeyPairEd25519.fromRandom();
+
+// create and sign a message
+const message = new Uint8Array(sha256.array('message'));
+const signature = keyPair.sign(message);
+
+// the message is verifiably signed by the correct key
+console.assert(keyPair.verify(message, signature.signature));
+```
+
+## Transaction signing
+
+### Life cycle of a transaction
+
+- **An app requests wallet to sign transaction(s)**  \
+  This is generally using wallet-specific protocols. The API used by NEAR Wallet is being defined in [NEP](https://github.com/nearprotocol/NEPs/pull/10).
+
+- **Wallet decodes [Borsh](http://borsh.io/)-serialized transactions and displays approval UI**  \
+  Schema defined [here](https://github.com/nearprotocol/nearlib/blob/8f5063bfee4ea7e7eba1f8dbfc20862534c0febf/src.ts/transaction.ts#L119)
+
+- **SHA-256 hash of transaction signed using ed25519**  \
+  For examples see [nearlib](https://github.com/nearprotocol/nearlib/blob/8f5063bfee4ea7e7eba1f8dbfc20862534c0febf/src.ts/transaction.ts#L198), [wallet-core](https://github.com/trustwallet/wallet-core/blob/951e73abfa0362b4d61202bac4e399a4faae97a8/src/NEAR/Signer.cpp#L20), [Ledger app](https://github.com/nearprotocol/near-ledger-app/blob/5abe5f5d57dff9cefe4535057d7a39f476d32d77/workdir/near-ledger-app/src/crypto/near.c#L7).
+
+- **Transaction sent to NEAR node**  \
+  This is a relatively simple [JSON-RPC call.](https://github.com/nearprotocol/nearlib/blob/8f5063bfee4ea7e7eba1f8dbfc20862534c0febf/src.ts/providers/json-rpc-provider.ts#L41)
+
+- **The App is notified**  \
+  With NEAR Wallet this is a simple redirect to the URL provided by the app when sending transaction.
+
+- **Transaction is available in block explorer**  \
+  Every transaction eventually will show up in [NEAR Explorer](http://explorer.nearprotocol.com) which can be also used by a wallet or exchange to query activity for given account.
+
+
+For additional detail around the life of a transaction on the NEAR platform please see this [Key Concepts: Transaction](/docs/concepts/transaction) page
+
+### Creating transaction offline
+
+Offline transaction signing in 3 steps:
+
+(1) Fetch latest block hash (online: requires access to pull data from the live network)
+
+```js
+const networkStatus = await near.connection.provider.status();
```

Comment:
first need to create near, e.g. something like `const near = new nearlib.Near();`

Reply:
haven't seen that before. will update after testing

Reply:
⏳

Reply:
for some reason when i read this i interpreted `new Near` as tokens and not the connection object.  for sure, will add

Reply:
✅ done